### DISCUSSION TITLE
brcmfmac wifi driver beta test

### DISCUSSION THREAD

**Discussion by bani6809** [X1Plus/X1Plus#102] [General]
beta test of https://github.com/X1Plus/X1Plus/pull/85 which integrates the brcmfmac wifi driver from newer kernels.

please smoke test this and let us know your results, especially if you are having wifi issues with the current wifi driver in oem or xplus 1.0!

<hr /><em>This discussion was created from the release <a href='https://github.com/X1Plus/X1Plus/releases/tag/'>brcmfmac wifi driver beta test</a>.</em>

**Comment by jwise** [comment]
wait this has the qrc changes that break kexec_ui, hoooold on a moment.

**Reply by bani6809** [reply]
i took current main which has the kexec_ui fixes and then merged this with it and made a build. i tested it, it works.

**Reply by jwise** [reply]
ahh I see. the tag was just my brcmfmac branch

**Comment by wattie2k1** [comment]
From a short test so far I can say that streaming from the cam to Orca Slicer is much more stable. On OEM firmware I had a lot of lags, this is now much better.
I tested transfering files files by sftp to sdcard on x1p 1.0 and this version, on x1p 1.0 I get solid transfer speeds from 1.5mb/s, on the test version I get stable speed of 2.5mb/s.
Besides that I didn't noticed any problems with the new driver, the printer stays connected on wifi and sending gcode files from slicer to printer through Bambu cloud works as expected too.

**Comment by bani6809** [comment]
dropped a new update 1.0-7+gca7e1d4.x1p built on @jwise's commit to update it to brcmfmac from v5.15.148

please test and verify no regressions

**Comment by wattie2k1** [comment]
Flashed successfully and wifi works as in the previous beta. Transfer speeds through sftp are also the same as before. The wifi indicator is still broken as in the previous beta. But for the rest I see no difference in speed and functionality as in the beta before.
